File: sangsoo/0x15 Hash/7785.cpp
#include <iostream>
#include <unordered_map>
#include <algorithm>
#include <vector>
#include <string>
using namespace std;
int n;
int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    cin >> n;
    unordered_map<string, string> m;
    for(int i = 0; i < n; i++){
        string name, el;
        cin >> name >> el;
        m[name] = el;
    }
    vector<string> v;
    for(auto e : m){
        if (e.second == "enter"){
            v.push_back(e.first);
        }
    }
    sort(v.rbegin(), v.rend()); // 역순 정렬
    for(auto e : v) cout << e << "\n";
}

File: sangsoo/0x15 Hash/7785.py
n = int(input())
m = {}
for i in range(n):
    name, el = input().split()
    m[name] = el
out = []
for k, v in m.items():
    if v == "enter":
        out.append(k)
out.sort(reverse = True)
for i in range(len(out)):
    print(out[i])